plugins/telemetry/telemetry.h

```diff
@@ -87,100 +141,458 @@ class Telemetry
+     */
     typedef std::function<void(bool in_air)> in_air_callback_t;
+
+    /**
+     * Subscribes to in-air updates (asynchronous).
+     *
+     * @param callback function to call with updates
+     */
     void in_air_async(in_air_callback_t callback);
 
+    /**
+     * Callback type for armed updates (asynchronous).
+     *
+     * @param armed true if armed (motors spinning)
+     */
     typedef std::function<void(bool armed)> armed_callback_t;
-    // Note: this function is limited to 1Hz.
+
+    /**
+     * Subscribes to armed updates (asynchronous).
+     *
+     * Note that armed updates are limited to 1Hz.
+     *
+     * @param callback function to call with updates
+     */
     void armed_async(armed_callback_t callback);
 
+    /**
+     * Callback type for attitude updates in quaternion.
+     *
+     * @param quaternion attitude quaternion
+     */
     typedef std::function<void(Quaternion quaternion)> attitude_quaternion_callback_t;
+
+    /**
+     * Subscribes to attitude updates in quaternion (asynchronous).
+     *
+     * @param callback function to call with updates
+     */
     void attitude_quaternion_async(attitude_quaternion_callback_t callback);
 
+    /**
+     * Callback type for attitude updates in Euler angles.
+     *
+     * @param euler_angle attitude Euler angle
+     */
     typedef std::function<void(EulerAngle euler_angle)> attitude_euler_angle_callback_t;
+
+    /**
+     * Subscribes to attitude updates in Euler angles (asynchronous).
+     *
+     * @param callback function to call with updates
+     */
     void attitude_euler_angle_async(attitude_euler_angle_callback_t callback);
 
+    /**
+     * Subscribes to camera attitude updates in quaternion (asynchronous).
+     *
+     * @param callback function to call with updates
+     */
     void camera_attitude_quaternion_async(attitude_quaternion_callback_t callback);
 
+    /**
+     * Subscribes to camera attitude updates in Euler angles (asynchronous).
+     *
+     * @param callback function to call with updates
+     */
     void camera_attitude_euler_angle_async(attitude_euler_angle_callback_t callback);
 
+    /**
+     * Callback type for ground speed (NED) updates.
+     *
+     * @param ground_speed_ned ground speed (NED)
+     */
     typedef std::function<void(GroundSpeedNED ground_speed_ned)> ground_speed_ned_callback_t;
+
+    /**
+     * Subscribes to ground speed (NED) updates (asynchronous).
+     *
+     * @param callback function to call with updates
+     */
     void ground_speed_ned_async(ground_speed_ned_callback_t callback);
 
+    /**
+     * Callback type for GPS information updates.
+     *
+     * @param gps_info GPS information
+     */
     typedef std::function<void(GPSInfo gps_info)> gps_info_callback_t;
+
+    /**
+     * Subscribes to GPS information updates (asynchronous).
+     *
+     * @param callback function to call with updates
+     */
```

Comment:
For all of these, is there any way to cancel async updates? Would anyone want to?

Reply:
Never thought about that. I guess no.